WalletWasabi.Fluent/ViewModels/AddWallet/Create/ConfirmRecoveryWordsViewModel.cs

```diff
@@ -9,63 +8,155 @@
-	[AutoNotify] private bool _isSkipEnable;
-
-	public ConfirmRecoveryWordsViewModel(
-		List<RecoveryWordViewModel> mnemonicWords,
-		Mnemonic mnemonic,
-		string walletName)
+	private readonly List<RecoveryWordViewModel> _words;
+	private readonly Mnemonic _mnemonic;
+	private readonly string _walletName;
+
+	[AutoNotify] private bool _isSkipEnabled;
+	[AutoNotify] private RecoveryWordViewModel? _currentWord;
+	[AutoNotify] private List<RecoveryWordViewModel> _availableWords;
+
+	public ConfirmRecoveryWordsViewModel(List<RecoveryWordViewModel> words, Mnemonic mnemonic, string walletName)
 	{
-		_confirmationWordsSourceList = new SourceList<RecoveryWordViewModel>();
-#if RELEASE
-		_isSkipEnable = Services.WalletManager.Network != Network.Main || System.Diagnostics.Debugger.IsAttached;
-#else
-		_isSkipEnable = true;
-#endif
-		var nextCommandCanExecute =
-			_confirmationWordsSourceList
+		_availableWords = new List<RecoveryWordViewModel>();
+		_words = words;
+		_mnemonic = mnemonic;
+		_walletName = walletName;
+	}
+
+	public ObservableCollectionExtended<RecoveryWordViewModel> ConfirmationWords { get; } = new();
+
+	[System.Diagnostics.CodeAnalysis.SuppressMessage("Reliability", "CA2000:Dispose objects before losing scope", Justification = "Uses DisposeWith()")]
+	protected override void OnNavigatedTo(bool isInHistory, CompositeDisposable disposables)
+	{
+		base.OnNavigatedTo(isInHistory, disposables);
+
+		ConfirmationWords.Clear();
+
+		var confirmationWordsSourceList = new SourceList<RecoveryWordViewModel>();
+
+		confirmationWordsSourceList
+			.DisposeWith(disposables)
 			.Connect()
 			.ObserveOn(RxApp.MainThreadScheduler)
-			.WhenValueChanged(x => x.IsConfirmed)
-			.Select(_ => _confirmationWordsSourceList.Items.All(x => x.IsConfirmed));
+			.Sort(SortExpressionComparer<RecoveryWordViewModel>.Ascending(x => x.Index))
+			.Bind(ConfirmationWords)
+			.OnItemAdded(x => x.Reset())
+			.Subscribe()
+			.DisposeWith(disposables);
 
 		EnableBack = true;
 
-		NextCommand = ReactiveCommand.CreateFromTask(() => OnNextAsync(mnemonic, walletName), nextCommandCanExecute);
+		CancelCommand = ReactiveCommand.Create(OnCancel);
 
-		if (_isSkipEnable)
+		var nextCommandCanExecute =
+			confirmationWordsSourceList
+			.Connect()
+			.WhenValueChanged(x => x.IsConfirmed)
+			.Select(_ => confirmationWordsSourceList.Items.All(x => x.IsConfirmed));
+
+		NextCommand = ReactiveCommand.CreateFromTask(OnNextAsync, nextCommandCanExecute);
+
+		SetSkip();
+
+		confirmationWordsSourceList.AddRange(_words);
+
+		AvailableWords =
+			confirmationWordsSourceList.Items
+									   .Select(x => new RecoveryWordViewModel(x.Index, x.Word))
+									   .OrderBy(_ => Random.Shared.Next())
+									   .ToList();
+
+		var availableWordsSourceList = new SourceList<RecoveryWordViewModel>();
+
+		availableWordsSourceList
+			.DisposeWith(disposables)
+			.Connect()
+			.WhenPropertyChanged(x => x.IsSelected)
+			.Subscribe(x => OnWordSelected(x.Sender));
```

Comment:
`OnWordSelected` -> `OnWordSelectionChanged` naming feels more telling as handles the deselection too.